.github/workflows/etl.yml

```diff
@@ -40,4 +40,4 @@ jobs:
       - name: Extract API docs
         env:
           EXTRACT_SANITY_API_TOKEN: ${{ secrets.EXTRACT_SANITY_API_TOKEN }}
-        run: yarn etl --quiet
+        run: yarn etl sanity
```

Comment:
The env variables are not set. I suggest adding a `.env` in the root of the monorepo with:

```sh
# Used by `yarn etl`
EXTRACT_SANITY_PROJECT_ID="<project-id-for-tsdoc>"
EXTRACT_SANITY_DATASET="<dataset-for-tsdoc>"
```

And then change the env variable names in `packages/sanity/tsdoc.config.ts` to:

```ts
import {defineConfig} from '@sanity/tsdoc'
import pkgConfig from './package.config'

export default defineConfig({
  extract: pkgConfig.extract,
  output: {
    sanity: {
      projectId: process.env.EXTRACT_SANITY_PROJECT_ID,
      dataset: process.env.EXTRACT_SANITY_DATASET,
      token: process.env.EXTRACT_SANITY_WRITE_TOKEN,
    },
  },
})
```

Reply:
@mariuslundgard thanks for this, it was very helpful!  
I managed to get it working, however, can you just double check that with the changes I've made (especially here https://github.com/sanity-io/sanity/blob/c34f20c3e149c891c5a1067992643f859c9be7bf/.github/workflows/etl.yml#L6) will make sure that it only runs when it's merged into `current`?

Because in our case we don't create a PR when we're releasing, it's a rebase I'm unsure if this would still trigger then? 🤔

Reply:
Yes, and you may also use an `if` condition to make sure it will not run if a PR were to be created mistakenly.

However – I’m wondering if what we want is to let it run on every PR, except for the _writing to Sanity_ part. That would be achieved by not including the token env variable. In that case, it will do the extraction and transformation, but not load it to Sanity. This could be a good way of testing that the API is extractable before releasing.

Reply:
> However – I’m wondering if what we want is to let it run on every PR, except for the writing to Sanity part. That would be achieved by not including the token env variable. In that case, it will do the extraction and transformation, but not load it to Sanity. This could be a good way of testing that the API is extractable before releasing.

I would say yes if we currently had a way of easily previewing the data. Because otherwise I'm not sure if there was a clear way to get feedback that it's not been "extracted properly"? I suppose the CI could just complain "I wasn't able to extract `X` api" and we wouldn't need to wait until the release for something to potentially go wrong... 🤔 

That could indeed be cool, let me see what I can do 🤔

Reply:
@mariuslundgard 
👋 I figured out something which works: https://github.com/sanity-io/sanity/blob/411195ad2922a73ed7ec4b495e6258daf2f66c9b/.github/workflows/etl.yml#L42-L44

There's basically a condition that checks if: you're running this on Push AND if the branch. If it's not in current (just doing the usual pushing to a PR) then it will use the `dev_`. otherwise it will use the other secret.
I haven't tested it on the "current" branch, but it does work with the dev information

**Why this over preventing it from uploading all together?** One of the topics we have been talking about is to have a connection with the website to more easily (visually) see the changes. What this means is that we will need to have a set up for a dev env at some point anyway. This feels better 👍

Reply:
Sweet! Next step is to figure out how we can get a render of whats in the dev dataset somewhere for us to have a preview (strictly speaking it doesn't even have to be the website as long as it can show us that it can render nicely, if that makes it easier somehow). But yeah, we can tackle that separately.

Great work @RitaDias!!! ✨